src/main/java/com/marmelab/quoridor/model/Board.java

```diff
@@ -52,26 +53,39 @@ public int getBoardSize() {
         return squares;
     }
 
-    public void addFence(final Fence fence) {
+    public void addFence(final Fence fence, final Pawn pawn) {
```

Comment:
Why are the fences part of the board but not the pawn?

Reply:
IMO: board, fences and pawn should be separate, just like in real life. Only the "game" links them together by its rules.